runtime-management/runtime-management-impl/src/main/java/io/cloudslang/runtime/impl/python/external/PythonExecutorLifecycleManagerImpl.java

```diff
@@ -0,0 +1,215 @@
+            doStartPythonExecutor();
+        }
+    }
+
+    private void doStartPythonExecutor() {
+        start();
+    }
+
+    @Override
+    public void start() {
+        logger.info("A request to start the Python Executor was sent");
+        if (isAlive()) {
+            logger.info("Python Executor is already running");
+            return;
+        }
+
+        destroyPythonExecutorProcess();
+
+        if (isWindows()) {
+            startWindowsProcess();
+        } else {
+            startLinuxProcess();
+        }
+
+        waitToStart();
+    }
+
+    @Override
+    public boolean isAlive() {
+        Response response;
+        try {
+            response = restEasyClient
+                    .target(EXTERNAL_PYTHON_EXECUTOR_URL)
+                    .path(EXTERNAL_PYTHON_EXECUTOR_HEALTH_PATH)
+                    .request()
+                    .accept(APPLICATION_JSON_TYPE)
+                    .header(CONTENT_TYPE, APPLICATION_JSON)
+                    .build("GET")
+                    .invoke();
+        } catch (Exception e) {
+            return false;
+        }
+
+        return response.getStatus() == 200;
+    }
+
+    @PreDestroy
+    @Override
+    public void stop() {
+        logger.info("A request to stop the Python Executor was sent");
+        if (!isAlive()) {
+            logger.info("Python Executor was already stopped");
+            return;
+        }
+
+        Response response = restEasyClient
+                .target(EXTERNAL_PYTHON_EXECUTOR_URL)
+                .path(EXTERNAL_PYTHON_EXECUTOR_STOP_PATH)
+                .request()
+                .accept(APPLICATION_JSON_TYPE)
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .header(AUTHORIZATION, ENCODED_AUTH)
+                .build("POST")
+                .invoke();
+
+        int statusCode = response.getStatus();
+        if (statusCode == 200) {
+            waitToStop();
+        } else {
+            logger.error("Failed to stop the Python Executor, response: " + statusCode);
+        }
+    }
+
+    private void startWindowsProcess() {
+        startProcess("start-python-executor.bat");
+    }
+
+    private void startLinuxProcess() {
+        startProcess("start-python-executor.sh");
+    }
+
+    private void startProcess(String startPythonExecutor) {
+        ProcessBuilder pb = new ProcessBuilder(PYTHON_EXECUTOR_BIN_PATH + separator + startPythonExecutor, PYTHON_EXECUTOR_PORT);
+        try {
+            logger.info("Starting Python Executor on port: " + PYTHON_EXECUTOR_PORT);
+            pythonExecutorProcess = pb.start();
+        } catch (IOException e) {
+            logger.error("Failed to start Python Executor", e);
+        }
+    }
+
+    private boolean isWindows() {
+        return SystemUtils.IS_OS_WINDOWS;
+    }
+
+    private void waitToStart() {
+        logger.info("Waiting to start");
+
+        for (int tries = 0; tries < 20; tries++) {
```

Comment:
20 looks like a magic number. Extract it to a suggestive named constant. Maybe even expose it as a SP:  private static final int START_STOP_RETRIES_COUNT = Integer.getInteger("python.executor.retriesCount", 20);